SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialStaticComponentView.h

```diff
@@ -15,22 +16,21 @@
 #include "SpatialStaticComponentView.generated.h"
 
 UCLASS()
-class SPATIALGDK_API USpatialStaticComponentView : public UObject
+class SPATIALGDK_API USpatialStaticComponentView : public UAbstractStaticComponentView
```

Comment:
It seems to not be difficult to use a real USpatialStaticComponentView in LockingPolicy test, instead of a Mock one. It would require creating a USpatialStaticComponentView and filling it with test data by calling `OnAuthorityChange` and `OnAddComponent` with some arguments you'd need it to return during the test. Would you still prefer creating `UAbstractStaticComponentView` and mocking it for tests?

Reply:
I took @ImprobableNic 's suggestion and changed the implementation to not have the `UAbstractStaticComponentView`. Instead there is a Mock which derives straight from `USpatialStaticComponentView` with an `Init(...)` method which calls the `OnAuthorityChange` and `OnAddComponent` appropriate as you suggest - does that sound good?

Reply:
That seems better in my opinion. What if we go even further, and remove `USpatialStaticComponentViewMock`, and create a free `Init()` function that constructs a `USpatialStaticComponentView` with test values?

Reply:
I think the deserved interface for the mock is being able to do trivial things like set HasAuthority to always return false, etc. I think that sort of interface should be contained in a mock centred around testing rather than exposing that on the actual implementation class.

Reply:
If you remove `UAbstractStaticComponentView` - you'd be able to put the template function back, thus reducing the PR size?